src/services/helpers.ts

```diff
@@ -62,10 +75,20 @@ export async function fetchWithValidator<T, O, I>(
   }
 
   const json = await response.json();
+  const statusCode = response.status;
+
   if (!response.ok) {
-    if (json.message === "Invalid authentication token provided")
+    const message = json.message ?? "Fetch failed";
+
+    if (json.message === "Invalid authentication token provided") {
       throw new SessionError(json.message);
-    throw new Error(json.message ?? "Fetch failed");
+    }
+
+    if (includeCodes && (json.errorCode || statusCode)) {
```

Comment:
Adding `includeCodes` toggle so we only return status code and error code for specific endpoints for compatibility.